### ISSUE TITLE
Only the first y axis displays the counts in ggpairs

### ISSUE THREAD

**Issue by rizbicki** [ggobi/ggally#259]
I'm running the following code:

```r
library(GGally)
data <- data.frame(x=runif(100),y=runif(100),z=runif(100))
ggpairs(data, 
    diag=list(continuous="barDiag"))
```

I get the following plot

![test](https://user-images.githubusercontent.com/10176589/36789304-d4f48748-1c6f-11e8-8aa7-2fdb3b058915.png)

The y axis of the top left panel is weird: it shows the counts on the histogram, and not the values of the variables. However, the y axis of the other variables are their values, and not the counts. Is this behavior expected?

Thanks

.

**Comment by schloerke** [comment]
@rizbicki 

Yes, this is expected behavior.  This is a side effect of each of the left column and bottom row plots displaying their individual axes.  With the ability to retrieve each plot individually, it does not make sense to arbitrarily change the axes of a plot to something that isn't what the plot is displaying (such as changing the counts to display a y range).  However, the correlation does not display anything of data, rather just text, so a grid can be added arbitrarily.  

To get around this unexpected behavior (but at the cost of losing the histograms), you can place the axes in the diagonal.

``` r
library(GGally)                            
ggpairs(iris, 1:3, axisLabels = "internal")
```

![](https://i.imgur.com/IRBgra6.png)

**Comment by schloerke** [comment]
(Closing as it is expected behavior). Feel free to comment if you have more questions!

**Comment by KunhuanLiu** [comment]
Hi,
I agree that the y axes should be pertinent to the graphs plotted, and it did not bother when I used upper triangle to show statistics.  However, when I use the upper triangle to show 2D density plots or scatterplots,  the first y-axis caused trouble and some confusion. I also noticed in Seaborn's [pairplot](https://seaborn.pydata.org/generated/seaborn.pairplot.html#seaborn.pairplot), they use first y-axis for y range.

I am wondering if there is any flexibility for users to add axes to the individual grid plots, if customizing for the edge y or x-axis is not possible.

**Comment by schloerke** [comment]
Copying in the image from Seaborn's pairplot:
![image](https://user-images.githubusercontent.com/93231/221297432-16d0fb30-597d-4971-ba98-9ba03b8b0466.png)

The top left plot has a y axis that represents the majority of the top row, but is attached to a plot that does not use that axis range. That plot uses an axis range of "count", not bill length in mm.

This feels really strange at first glance.

--------------------

Seaborn's approach does provide a better axis to `k-1` plots, rather than just the immediate (single) plot.  

@dicook What are you thoughts on switching the default axis label in the top left and bottom right to the "top left" and "bottom right" axes of the upper triangle? (Iff there are lower triangle plots. Iff `axisLabels != "internal"`)

![Screenshot 2023-02-24 at 4 48 21 PM](https://user-images.githubusercontent.com/93231/221299938-10e9aed3-09d6-40cd-91a8-daeea49cbbe5.png)

**Comment by KunhuanLiu** [comment]
I completely agree that to use length in nm as axis where the immediate plot is a histogram can be misleading.  On the other hand, when upper triangle region do have scatter plots / density 2d plots, audience may be interested to read off the axes.  

I  wonder if placing the physical-unit axes on shown places is something I can do at this moment (without having you to change the default behavior):
1) show a y-axis on the 2nd plot where axis in physical unit is preferred to read graphs (knowing that there will be limited space, so axis like "1000, 2000, 3000,...6000" wouldn't fit)
2) show a y-axis in physical unit on the top right hand side for the upper triangles. Turn off the count axis.

![221297432-16d0fb30-597d-4971-ba98-9ba03b8b0466](https://user-images.githubusercontent.com/22421218/221324763-0f206b42-a5bc-48fd-b6e7-1ffe93cb93fe.png)

Related thought:  Histograms in the diagonal don't have count axes either.  I doubt people would really care, and if the counts do matter, it should be able to be turned on for any diagonal histograms; not sure how to do it at the moment.

Thanks for the timely response and all the work,

**Comment by dicook** [comment]
I think we were originally motivated by ggplot facets, but I do agree that the left and bottom could be a better default.Sent from my iPhoneOn 25 Feb 2023, at 11:32 am, Kunnn ***@***.***> wrote:﻿
I completely agree that to use length in nm as axis where the immediate plot is a histogram can be misleading.  On the other hand, when upper triangle region do have scatter plots / density 2d plots, audience may be interested to read off the axes.
I  wonder if placing the physical-unit axes on shown places is something I can do at this moment (without having you to change the default behavior):

show a y-axis on the 2nd plot where axis in physical unit is preferred to read graphs (knowing that there will be limited space, so axis like "1000, 2000, 3000,...6000" wouldn't fit)
show a y-axis in physical unit on the top right hand side for the upper triangles. Turn off the count axis.


Related thought:  Histograms in the diagonal don't have count axes either.  I doubt people would really care, and if the counts do matter, it should be able to be turned on for any diagonal histograms; not sure how to do it at the moment.
Thanks for the timely response and all the work,
